swarm/src/lib.rs

```diff
@@ -407,8 +407,8 @@ where TBehaviour: NetworkBehaviour<ProtocolsHandler = THandler>,
     }
 
     /// Returns the peer ID of the swarm passed as parameter.
-    pub fn local_peer_id(me: &Self) -> &PeerId {
-        &me.network.local_peer_id()
+    pub fn local_peer_id(me: &Self) -> PeerId {
```

Comment:
Should we treat API changes like these as breaking changes?

Reply:
Strictly speaking, yes, since it is a change of return type. We could of course keep having `Network::local_peer_id` and `Swarm::local_peer_id` return `&PeerId`, I may have done an unnecessary API change here.

Reply:
I undid this unnecessary API change in https://github.com/libp2p/rust-libp2p/pull/1950/commits/db68488549e2f99e61919f2c8c93b17c68a8d41f. Thanks for bringing it up.